File: com-mini-spring-core/src/main/java/com/mini/digest/Base64Util.java
package com.mini.digest;

import java.nio.ByteBuffer;
import java.util.Base64;

/**
 * Base64Util.java
 * @author xchao
 */
public class Base64Util {

    /**
     * Base64编码
     * @param data 编码数据
     * @return 编码结果
     */
    public static String encode(String data) {
        return Base64.getEncoder().encodeToString(data.getBytes());
    }

    /**
     * Base64编码
     * @param data 编码数据
     * @return 编码结果
     */
    public static byte[] encodeToByte(String data) {
        return Base64.getEncoder().encode(data.getBytes());
    }

    /**
     * Base64编码
     * @param data 编码数据
     * @return 编码结果
     */
    public static String encode(byte[] data) {
        return Base64.getEncoder().encodeToString(data);
    }

    /**
     * Base64编码
     * @param data 编码数据
     * @return 编码结果
     */
    public static byte[] encodeToByte(byte[] data) {
        return Base64.getEncoder().encode(data);
    }

    /**
     * Base64编码
     * @param data 编码数据
     * @return 编码结果
     */
    public static ByteBuffer encode(ByteBuffer data) {
        return Base64.getEncoder().encode(data);
    }

    /**
     * Base64编码
     * @param data 编码数据
     * @return 编码结果
     */
    public static int encode(byte[] data, byte[] dest) {
        return Base64.getEncoder().encode(data, dest);
    }

    /**
     * Base64解码
     * @param data 编码数据
     * @return 编码结果
     */
    public static String decode(String data) {
        return new String(Base64.getDecoder().decode(data));
    }

    /**
     * Base64解码
     * @param data 编码数据
     * @return 编码结果
     */
    public static byte[] decodeToByte(String data) {
        return Base64.getDecoder().decode(data);
    }

    /**
     * Base64解码
     * @param data 编码数据
     * @return 编码结果
     */
    public static String decode(byte[] data) {
        return new String(Base64.getDecoder().decode(data));
    }

    /**
     * Base64解码
     * @param data 编码数据
     * @return 编码结果
     */
    public static byte[] decodeToByte(byte[] data) {
        return Base64.getDecoder().decode(data);
    }

    /**
     * Base64解码
     * @param data 编码数据
     * @return 编码结果
     */
    public static ByteBuffer decode(ByteBuffer data) {
        return Base64.getDecoder().decode(data);
    }

    /**
     * Base64解码
     * @param data 编码数据
     * @param dest 编码参数
     * @return 编码结果
     */
    public static int decode(byte[] data, byte[] dest) {
        return Base64.getDecoder().decode(data, dest);
    }

    /**
     * URLBase64编码
     * @param data 编码数据
     * @return 编码结果
     */
    public static String urlEncode(String data) {
        return Base64.getUrlEncoder().encodeToString(data.getBytes());
    }

    /**
     * URLBase64编码
     * @param data 编码数据
     * @return 编码结果
     */
    public static byte[] urlEncodeToByte(String data) {
        return Base64.getUrlEncoder().encode(data.getBytes());
    }

    /**
     * URLBase64编码
     * @param data 编码数据
     * @return 编码结果
     */
    public static String urlEncode(byte[] data) {
        return Base64.getUrlEncoder().encodeToString(data);
    }

    /**
     * URLBase64编码
     * @param data 编码数据
     * @return 编码结果
     */
    public static byte[] urlEncodeToByte(byte[] data) {
        return Base64.getUrlEncoder().encode(data);
    }

    /**
     * URLBase64编码
     * @param data 编码数据
     * @return 编码结果
     */
    public static ByteBuffer urlEncode(ByteBuffer data) {
        return Base64.getUrlEncoder().encode(data);
    }

    /**
     * URLBase64编码
     * @param data 编码数据
     * @return 编码结果
     */
    public static int urlEncode(byte[] data, byte[] dest) {
        return Base64.getUrlEncoder().encode(data, dest);
    }

    /**
     * URLBase64解码
     * @param data 解码数据
     * @return 解码结果
     */
    public static String urlDecode(String data) {
        return new String(Base64.getUrlDecoder().decode(data));
    }

    /**
     * URLBase64解码
     * @param data 解码数据
     * @return 解码结果
     */
    public static byte[] urlDecodeToByte(String data) {
        return Base64.getUrlDecoder().decode(data);
    }

    /**
     * URLBase64解码
     * @param data 解码数据
     * @return 解码结果
     */
    public static String urlDecode(byte[] data) {
        return new String(Base64.getUrlDecoder().decode(data));
    }

    /**
     * URLBase64解码
     * @param data 解码数据
     * @return 解码结果
     */
    public static byte[] urlDecodeToByte(byte[] data) {
        return Base64.getUrlDecoder().decode(data);
    }

    /**
     * URLBase64解码
     * @param data 解码数据
     * @return 解码结果
     */
    public static ByteBuffer urlDecode(ByteBuffer data) {
        return Base64.getUrlDecoder().decode(data);
    }

    /**
     * URLBase64解码
     * @param data 解码数据
     * @param dest 编码参数
     * @return 解码结果
     */
    public static int urlDecode(byte[] data, byte[] dest) {
        return Base64.getUrlDecoder().decode(data, dest);
    }
}

File: sn.mini.kotlin.util/src/main/java/sn/mini/kotlin/util/degest/Base64Util.kt
package sn.mini.kotlin.util.degest

import java.nio.ByteBuffer

object Base64Util {

    /**
     * Base64编码
     * @param data
     * @return
     */
    fun encode(data: String): String {
        return java.util.Base64.getEncoder().encodeToString(data.toByteArray())
    }

    /**
     * Base64编码
     * @param data
     * @return
     */
    fun encodeToByte(data: String): ByteArray {
        return java.util.Base64.getEncoder().encode(data.toByteArray())
    }

    /**
     * Base64编码
     * @param data
     * @return
     */
    fun encode(data: ByteArray): String {
        return java.util.Base64.getEncoder().encodeToString(data)
    }

    /**
     * Base64编码
     * @param data
     * @return
     */
    fun encodeToByte(data: ByteArray): ByteArray {
        return java.util.Base64.getEncoder().encode(data)
    }

    /**
     * Base64编码
     * @param data
     * @return
     */
    fun encode(data: ByteBuffer): ByteBuffer {
        return java.util.Base64.getEncoder().encode(data)
    }

    /**
     * Base64编码
     * @param data
     * @return
     */
    @JvmStatic
    fun encode(data: ByteArray, dest: ByteArray): Int {
        return java.util.Base64.getEncoder().encode(data, dest)
    }

    /**
     * Base64解码
     * @param data
     * @return
     */
    fun decode(data: String): String {
        return String(java.util.Base64.getDecoder().decode(data))
    }

    /**
     * Base64解码
     * @param data
     * @return
     */
    fun decodeToByte(data: String): ByteArray {
        return java.util.Base64.getDecoder().decode(data)
    }

    /**
     * Base64解码
     * @param data
     * @return
     */
    fun decode(data: ByteArray): String {
        return String(java.util.Base64.getDecoder().decode(data))
    }

    /**
     * Base64解码
     * @param data
     * @return
     */
    fun decodeToByte(data: ByteArray): ByteArray {
        return java.util.Base64.getDecoder().decode(data)
    }

    /**
     * Base64解码
     * @param data
     * @return
     */
    fun decode(data: ByteBuffer): ByteBuffer {
        return java.util.Base64.getDecoder().decode(data)
    }

    /**
     * Base64解码
     * @param data
     * @return
     */
    fun decode(data: ByteArray, dest: ByteArray): Int {
        return java.util.Base64.getDecoder().decode(data, dest)
    }

    /**
     * URLBase64编码
     * @param data
     * @return
     */
    fun urlEncode(data: String): String {
        return java.util.Base64.getUrlEncoder().encodeToString(data.toByteArray())
    }

    /**
     * URLBase64编码
     * @param data
     * @return
     */
    fun urlEncodeToByte(data: String): ByteArray {
        return java.util.Base64.getUrlEncoder().encode(data.toByteArray())
    }

    /**
     * URLBase64编码
     * @param data
     * @return
     */
    fun urlEncode(data: ByteArray): String {
        return java.util.Base64.getUrlEncoder().encodeToString(data)
    }

    /**
     * URLBase64编码
     * @param data
     * @return
     */
    fun urlEncodeToByte(data: ByteArray): ByteArray {
        return java.util.Base64.getUrlEncoder().encode(data)
    }

    /**
     * URLBase64编码
     * @param data
     * @return
     */
    fun urlEncode(data: ByteBuffer): ByteBuffer {
        return java.util.Base64.getUrlEncoder().encode(data)
    }

    /**
     * URLBase64编码
     * @param data
     * @return
     */
    fun urlEncode(data: ByteArray, dest: ByteArray): Int {
        return java.util.Base64.getUrlEncoder().encode(data, dest)
    }

    /**
     * URLBase64解码
     * @param data
     * @return
     */
    fun urlDecode(data: String): String {
        return String(java.util.Base64.getUrlDecoder().decode(data))
    }

    /**
     * URLBase64解码
     * @param data
     * @return
     */
    fun urlDecodeToByte(data: String): ByteArray {
        return java.util.Base64.getUrlDecoder().decode(data)
    }

    /**
     * URLBase64解码
     * @param data
     * @return
     */
    fun urlDecode(data: ByteArray): String {
        return String(java.util.Base64.getUrlDecoder().decode(data))
    }

    /**
     * URLBase64解码
     * @param data
     * @return
     */
    fun urlDecodeToByte(data: ByteArray): ByteArray {
        return java.util.Base64.getUrlDecoder().decode(data)
    }

    /**
     * URLBase64解码
     * @param data
     * @return
     */
    fun urlDecode(data: ByteBuffer): ByteBuffer {
        return java.util.Base64.getUrlDecoder().decode(data)
    }

    /**
     * URLBase64解码
     * @param data
     * @return
     */
    fun urlDecode(data: ByteArray, dest: ByteArray): Int {
        return java.util.Base64.getUrlDecoder().decode(data, dest)
    }
}
